src/Shared/MemberInfoExtensions.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.EntityFrameworkCore.Internal;
+
+namespace System.Reflection
+{
+    internal static class MemberInfoExtensions
+    {
+        public static Type GetMemberType(this MemberInfo memberInfo)
+            => (memberInfo as PropertyInfo)?.PropertyType ?? ((FieldInfo)memberInfo)?.FieldType;
+
+        public static bool IsSameAs(this MemberInfo propertyInfo, MemberInfo otherPropertyInfo)
+            => propertyInfo == null
+                ? otherPropertyInfo == null
+                : (otherPropertyInfo == null
+                    ? false
+                    : Equals(propertyInfo, otherPropertyInfo)
+                      || (propertyInfo.Name == otherPropertyInfo.Name
+                          && (propertyInfo.DeclaringType == otherPropertyInfo.DeclaringType
+                              || propertyInfo.DeclaringType.GetTypeInfo().IsSubclassOf(otherPropertyInfo.DeclaringType)
+                              || otherPropertyInfo.DeclaringType.GetTypeInfo().IsSubclassOf(propertyInfo.DeclaringType)
+                              || propertyInfo.DeclaringType.GetTypeInfo().ImplementedInterfaces.Contains(otherPropertyInfo.DeclaringType)
+                              || otherPropertyInfo.DeclaringType.GetTypeInfo().ImplementedInterfaces.Contains(propertyInfo.DeclaringType))));
```

Comment:
I'm always up for some expression-bodied ternary operators....but this one might be a bit much.

Your call, but just something to think about.

Reply:
This file was copied as-is from EF Core, so let's keep it like that for parity.